Sources/StreamChatSwiftUI/ChatChannel/ChatChannelDataSource.swift

```diff
@@ -161,7 +161,11 @@ class MessageThreadDataSource: ChannelDataSource, ChatMessageControllerDelegate
     weak var delegate: MessagesDataSource?
     
     var messages: LazyCachedMapCollection<ChatMessage> {
-        messageController.replies
+        var replies = messageController.replies
+        if let message = messageController.message, replies.last != message {
+            replies.append(message)
+        }
+        return replies
```

Comment:
This fixed an issue where showing a sheet and dismissing it, triggered channel view model to read data source's messages which dropped the parent message. 
e.g. use thread reply for polls, in the thread, open a sheet > poll disappears